examples/improved_wgan.py

```diff
@@ -72,7 +73,7 @@ def gradient_penalty_loss(y_true, y_pred, trainable_weights):
     y_true and y_pred to loss functions. To get around this, we make a partial() of the function with the
     trainable_weights argument, and use that for model training."""
     averaged_samples_loss = K.mean(y_pred)
-    gradients = K.gradients(averaged_samples_loss, trainable_weights)
+    gradients = K.gradients(averaged_samples_loss, averaged_samples)

```

Comment:
You're right, though I think using K.mean() might just result in constant scaling of the gradient compared to the correct version? (Specifically scaling it down by BATCH_SIZE). Still, I'm not sure the mathematical world is ready for BATCH_SIZE-Lipschitz functions just yet, so I'll make the change :p